### PR TITLE
Final Capstone Project: Car Test Drive Booking System Merge Request

### PR DESCRIPTION
**Pull request by ClaudiaRojasSoto**
## Overview

This pull request is a formal request to merge the cumulative and concerted efforts of our development team from the `develop` branch into the `master` branch. Our Final Capstone Project is a sophisticated car test drive booking system, designed and tailored following the prototype provided by Murat Korkmaz on Behance, with our unique spin - instead of motorcycles, we facilitate the booking of car test drives.

Our development team, composed of myself and Pablo Bonasera, has meticulously adhered to coding best practices, proper Gitflow, and comprehensive professional documentation throughout the project lifecycle.

Pull Request from front-end:
https://github.com/ClaudiaRojasSoto/Final-Capstone-Front-End/pull/15

### General Requirements Met

- Zero linter errors across the codebase.
- Strict adherence to Gitflow practices.
- Professional-level documentation and code commenting.

#### Chosen Theme

- Our application is built around the concept of booking test drives for cars, aiming to streamline the process of scheduling a drive with an intuitive user interface.

#### Core Features Implemented

- Simplified user login system by typing only the username.
- The navigation panel provides seamless access to:
  - Car listings for test drives.
  - Test drive reservation form.
  - User’s test drive reservations.
  - Functionality to add new cars to the system.
  - Option to remove cars from the system.
- The main page features a listing of cars available for test drives.
- Detail pages for each car, with an option for users to reserve a test drive.
- A fully responsive design, with tailored user experiences for both mobile and desktop versions.

#### Additional Features (Beyond Basic Requirements)

- We also integrated the ability for users to delete car listings, empowering administrative management within the app.
- The reservation process is enhanced by allowing users to select a date and city for their test drive, with auto-filling of the user's name and the selected car when the reservation is made through the car's detail page.

### Comprehensive Documentation

- Detailed API documentation has been included to facilitate understanding and potentially scaling the application in the future.

### Merge Request Justification

With all required features meticulously implemented, as well as the inclusion of additional capabilities not mandated by our team size, we are confident that our project not only fulfills but surpasses the specified criteria. We kindly request the review and approval for the merging of our `develop` branch into `master`.

We eagerly await any feedback and are fully prepared to conduct further modifications if recommended by our esteemed reviewers.

Thank you for your consideration and time.

Warm regards,
@PabloBona 
@ClaudiaRojasSoto

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+<div align="center">
```
Comment (Meltrust) [inline README.md:1]: - [x] Kindly, add a "Tests" section on your readme with specific instructions about how to correctly run your tests. 👍

Comment (Meltrust) [inline README.md:1]: - [x]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

Comment (Meltrust) [inline README.md:1]: - [x] Kindly, include in your instructions the required db commands that your users will need to run your backend app: ``rails db:create, rails db:migrate``. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @ClaudiaRojasSoto,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
#### Hi @ClaudiaRojasSoto,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme has been improved✔️ 
- Great job✔️ 




Cheers and Happy coding!👏👏👏


------

**Comment by ClaudiaRojasSoto** [comment]
Thank you so much @Meltrust.

**Pull request merged**